### ISSUE TITLE
ReadOnlyEngineTests testSearcherId fails

### ISSUE THREAD

**Issue by mayya-sharipova** [elastic/elasticsearch#66869]
**Build scan**:

https://gradle-enterprise.elastic.co/s/j4tztllphmnpo

**Repro line**:
./gradlew ':server:test' --tests "org.elasticsearch.index.engine.ReadOnlyEngineTests.testSearcherId" \
  -Dtests.seed=5F23C74BF9BA02D2 \
  -Dtests.security.manager=true \
  -Dtests.locale=es-AR \
  -Dtests.timezone=America/Noronha \
  -Druntime.java=11

**Reproduces locally?**:
yes on 7.x;   no failure on master (not in CI either)

**Applicable branches**:
only 7.x

**Failure history**:
14 failures over the last week

**Failure excerpt**:
```
org.elasticsearch.index.engine.ReadOnlyEngineTests > testSearcherId FAILED
    java.lang.AssertionError:
    Expected: not "0b4176a0054fce403abb72350d27706d167ec061d30031a49e5db37e4a50113b"
         but: was "0b4176a0054fce403abb72350d27706d167ec061d30031a49e5db37e4a50113b"
        at __randomizedtesting.SeedInfo.seed([5F23C74BF9BA02D2:BDA0039B9DAD7920]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.index.engine.ReadOnlyEngineTests.testSearcherId(ReadOnlyEngineTests.java:373)
```

**Comment by ywangd** [comment]
Another one for 7.x: https://gradle-enterprise.elastic.co/s/5iuefv3btss72

```
java.lang.AssertionError: |  
-- | --
  | Expected: not "45be586f712ec20585769456283fa5c4c17cf1d080579342d34c266754a2c10d" |  
  | but: was "45be586f712ec20585769456283fa5c4c17cf1d080579342d34c266754a2c10d"
```

Reproducible locally with:
> ./gradlew ':server:test' --tests "org.elasticsearch.index.engine.ReadOnlyEngineTests.testSearcherId" -Dtests.seed=70ADB7A5B6AE145E -Dtests.security.manager=true -Dtests.locale=id-ID -Dtests.timezone=UCT -Druntime.java=8

The failure is seed specific, so I am *not* muting it.

**Comment by dnhatn** [comment]
Fixed in #66963
